src/luarocks/build/builtin.lua

```diff
@@ -240,10 +267,28 @@ function builtin.run(rockspec)
             if not ok then return nil, err end
          end
          lib_modules[module_name] = dir.path(libdir, module_name)
-         ok = compile_library(module_name, objects, info.libraries, info.libdirs, name)
+         ok = compile.dynamic_library(module_name, objects, info.libraries, info.libdirs, name)
          if not ok then
             return nil, "Failed compiling module "..module_name
          end
+
+         if flags["static"] then
```

Comment:
Need to fail gracefully on platforms where static libraries are currently unsupported (e.g. win32)

Reply:
should I extend this condition of platforms? or just after `if not ok then` condition?

Reply:
After this line I would put something like: `if compile.static_object == nil then return nil, "static library compilation not available for your platform" end`
You'd then need to fix up which platforms default to static compilation on.